pkg/openstack/instances.go

```diff
@@ -633,6 +635,34 @@ func nodeAddresses(srv *servers.Server, ports []ports.Port, networkingOpts Netwo
 		return nil, err
 	}
 
+	// Add the addresses assigned on subports via trunk
+	// This exposes the vlan networks to which subports are attached
+	if client != nil {
+		subports, err := getSubports(client, srv.ID)
```

Comment:
Note that we're calling getSubports unconditionally for everybody here, but we're *not* passing in the list of ports that we already fetched, which already contains the subports.